datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
Interesting. This is not the behavior I expected from reading the documentation, but based on the PostgreSQL [transaction isolation](https://www.postgresql.org/docs/current/static/transaction-iso.html) documentation, it makes sense, since it's locking the table for the insert and waiting on concurrent transactions to commit or rollback. My assumption was that was that didn't block and resolved itself at commit-time.

However, I still fail to understand why you're not committing the transaction.